cogito/putter.go

```diff
@@ -262,8 +273,9 @@ func checkGitRepoDir(dir, owner, repo string) error {
 	if err != nil {
 		return fmt.Errorf(".git/config: remote: %w", err)
 	}
-	left := []string{"github.com", owner, repo}
```

Comment:
Repositories hosted on GitHub enterprise have unpredictable "hostnames" in their git config. I couldn't think of a way to generically test what the value of the field should be, and I couldn't think about why we would need to test on this now that we support a broader set of GitHub hosts. So, I removed it the hostname validation part and kept the rest. 

Is this okay?

Reply:
Just a bit of background here first. This part has two purposes:
- ensure that the git resource directory given as input to cogito matches the one configured in cogito source configuration. We must ensure that the information parsed "locally" from the git directory matches the information configured in the source. That means all: endpoint, organization, and repository name
- give the best possible information to the end user. People usually copy-paste the Concourse YAML configuration from other existing pipelines. That means that they copy-paste the cogito source configuration, and while they change the majority of the fields sometimes they forget some or make a typo.
The error msg like should ensure that the mistake is easily and quickly identified:
```
The received git repository is incompatible with the Cogito configuration.
Git repository configuration (received as 'inputs:' in this PUT step):
 url: git@<input-hostname>:<input-org-name>/<input-repo-name>.git
 owner: <input-org-name>
 repo: <input-repo-name>
Cogito SOURCE configuration:
 owner: <source-org-name>
 repo: <source-repo-name>
```

So instead of removing one check for the endpoint, you now actually MUST perform this check because if one configures `github_api_endpoint` in source configuration, you would expect this to match to what you parse from your local `.git/config`. 

To summarize, one must validate that:
- <input-hostname> == <source-github_api_endpoint>
- <input-org-name> == <source-org-name>
- <source-repo-name> == <source-repo-name>